mk/compile.mk

```diff
@@ -120,7 +120,7 @@ comp-cppflags-$2 = $$(filter-out $$(CPPFLAGS_REMOVE) $$(cppflags-remove) \
 		      $$(addprefix -I,$$(incdirs-$2)) \
 		      $$(cppflags$$(comp-sm-$2)) \
 		      $$(cppflags-lib$$(comp-lib-$2)) $$(cppflags-$2)) \
-		      -D__FILE_ID__=$$(subst -,_,$$(subst /,_,$$(subst .,_,$1)))
+		      -D__FILE_ID__=$$(subst -,_,$$(subst /,_,$$(subst .,_,$$(notdir $1))))
```

Comment:
`__FILE_ID__` must be unique in the source tree, so just the file name isn't enough. We need the path starting from optee_os.git.

Reply:
I think it's a fair requirement that source file names in a project need to be unique for each build target.

But I can try to setup a filter. The problem looks like this:

```
~/src/base/build/tmp_zynqmp-kria-starter/work/zynqmp_kria_starter-poky-linux/optee-os/4.2.0/git# strings ../image/lib/firmware/tee.elf | grep mikko.rapeli
__FILE_ID__ _home_mikko_rapeli_src_base_build_tmp_zynqmp_kria_starter_work_zynqmp_kria_starter_poky_linux_optee_os_4_2_0_build_core_ta_pub_key_c
__FILE_ID__ _home_mikko_rapeli_src_base_build_tmp_zynqmp_kria_starter_work_zynqmp_kria_starter_poky_linux_optee_os_4_2_0_build_core_ldelf_hex_c
__FILE_ID__ _home_mikko_rapeli_src_base_build_tmp_zynqmp_kria_starter_work_zynqmp_kria_starter_poky_linux_optee_os_4_2_0_build_core_early_ta_fd02c9da_306c_48c7_a49c_bbd827ae86ee_c
```

Reply:
Another alternative would be to provide generated source file names to __FILE_ID__ with relative paths, like normal files. Did not figure out where this is done yet.